source/common/network/transport_socket_options_impl.h

```diff
@@ -92,7 +92,7 @@ class TransportSocketOptionsUtility {
   fromFilterState(const StreamInfo::FilterStateSharedPtr& stream_info);
 };
 
-class CommonTransportSocketFactory : public TransportSocketFactory {
+class CommonUpstreamTransportSocketFactory : public UpstreamTransportSocketFactory {
```

Comment:
Just merge this interface with UpstreamTransportSocketFactory? It existed because we didn't have upstream/downstream transport socket factory split.

Reply:
This one has an implementation of `hashKey` so it will make `UpstreamTransportSocketFactory` not a pure interface. Is that OK with core interface carrying hashing implementation?

Reply:
Oh ok, nvm.